firmware/application/hw/debounce.cpp

```diff
@@ -52,84 +52,99 @@ bool Debounce::long_press_occurred() {
 bool Debounce::feed(const uint8_t bit) {
```

Comment:
Future: The logic is getting a bit hard to follow because of the mixed modes. Do you think there's be a lot of redundant code if you were to make 2 functions (one for normal, one for long press)? Or we had kind talked about maybe passing more raw info up to another layer where different "handler/filters" could be applied.

Reply:
There could be some benefit in doing that.  The main purpose of the original function was to debounce the switches, after which the code could potentially go to a different function if repeat is enabled, long press is enabled, or neither.  I'll think about it for a future PR.

Reply:
I've been thinking about it for a while and the downside of a "filter" layer is that we don't get good timing info at other layers. I had thought that keeping this layer strictly debounce (answers the question is down/up) and moving the decision about whether to repeat/long press/etc to higher layer -- it still has to all be driven though the timer ISR otherwise. I'm just kind of thinking out loud here, no solutions.